dags/oss_know/libs/email_address/init_email_address_from_ck.py

```diff
@@ -0,0 +1,179 @@
+import datetime
+
+from oss_know.libs.util.clickhouse_driver import CKServer
+from oss_know.libs.util.base import infer_country_from_emailcctld, infer_country_from_emaildomain, \
+    infer_company_from_emaildomain, infer_country_from_company
+
+
+def load_all_email_address(clickhouse_server_info):
+    ck = CKServer(host=clickhouse_server_info["HOST"],
+                  port=clickhouse_server_info["PORT"],
+                  user=clickhouse_server_info["USER"],
+                  password=clickhouse_server_info["PASSWD"],
+                  database=clickhouse_server_info["DATABASE"])
+
+    gits_sql = "SELECT DISTINCT author_email,committer_email FROM gits"
+    gits_email_from_ck = ck.execute_no_params(gits_sql)
+    all_email_address_dict = {}
+    for author_email, committer_email in gits_email_from_ck:
+        for item in author_email, committer_email:
+            if item:
+                all_email_address_dict[item] = 0
+    if all_email_address_dict:
+        for k, v in all_email_address_dict.items():
+            github_committer_id_from_committer_email_sql = f"SELECT DISTINCT author__id,commit__author__email from github_commits WHERE commit__author__email = '{k}'"
+            github_committer_id_from_committer_email = ck.execute_no_params(
+                github_committer_id_from_committer_email_sql)
+            github_author_id_from_author_email_sql = f"SELECT DISTINCT committer__id,commit__committer__email from github_commits WHERE commit__committer__email = '{k}'"
+            github_author_id_from_author_email = ck.execute_no_params(github_author_id_from_author_email_sql)
+            for item in github_committer_id_from_committer_email, github_author_id_from_author_email:
+                if item and item[0][0]:
+                    all_email_address_dict[item[0][1]] = item[0][0]
+
+    github_profile_sql = "SELECT DISTINCT email,id FROM github_profile"
+    github_profile_email_from_ck = ck.execute_no_params(github_profile_sql)
+    for email, id in github_profile_email_from_ck:
+        if email:
+            all_email_address_dict[email] = id
+
+    # count = 0
+    insert_email_address_data_params = []
+    if all_email_address_dict:
+        for k, v in all_email_address_dict.items():
+            insert_email_address_data = []
```

Comment:
Check if it's possible to insert a key-value pair

Reply:
Done,Thanks:)